### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by crew102**
In combination with commits f8e89335f07ee18816c2f15a07f94255e7a22f95, dbe4589094691d58ca7cca7d81858f9432fbb727, and 449bdc53fd144498e6f42e66d285e3d84fa631ab, this PR should lead to passing tests on travis (closing #69). Master was not passing before this branch b/c the name of one of the the test figures didn't match the name of the corresponding function in `funList`.

### CODE REVIEW

**Comment by crew102** [comment]
baaah, guess not. I'll take a closer look why later this week.

**Comment by daattali** [comment]
Sorry travis failed us and all your work :(

**Comment by crew102** [comment]
her tricks won't last much longer...

**Comment by daattali** [comment]
I think Travis is more commonly a male name...

File: R/ggExtra.R [hunk 1]
```diff
@@ -9,4 +9,4 @@
 #'
 #' @docType package
 #' @name ggExtra
-NULL
+NULL
```
Comment (daattali) [inline R/ggExtra.R]: can you set your editor to not remove newlines at end of files?

File: tests/testthat/helper-funs.R [hunk 2]
```diff
@@ -40,30 +40,105 @@ funList <-
...
+  
+  unloadPackages(packages = names(packageVersions))
+  on.exit(unloadPackages(packages = names(packageVersions)))
+  
   dir <- tempfile()
   dir.create(dir)
   on.exit(unlink(dir))
-  withr::with_libpaths(dir, action = "prefix", {
-    on.exit(unloadNamespace("ggplot2"))
+  names(packageVersions)
```
Comment (daattali) [inline tests/testthat/helper-funs.R]: this line and the next don't seem to do anything?

File: R/ggExtra.R (hunk 1 above)
Comment (crew102) [reply R/ggExtra.R]: I removed these lines manually. Some of the files had blank lines at the EOF while others didn't, so I was trying to make it consistent. Do you prefer to have blank lines at EOF?

File: tests/testthat/helper-funs.R (hunk 2 above)
Comment (crew102) [reply tests/testthat/helper-funs.R]: Yes, they should be removed.

**Comment by crew102** [comment]
I'm thinking we tell travis to build in a docker container (I believe travis supports this). Are you ok with that?

File: R/ggExtra.R (hunk 1 above)
Comment (daattali) [reply R/ggExtra.R]: blank line at EOF is preferred. I don't actively go around different files adding it, but I prefer not to make any files go the opposite direction

**Comment by daattali** [comment]
Yes building in a docker container is fine. As long as the tests run and do their job of testing :)

**Comment by crew102** [comment]
This PR should *finally* close #69.  The functional changes include:

a89c53f - The names of the test figures double as the names of the functions that are used to render the figures at test time. One of the names was incorrect, resulting in R not finding the plotting function. 

686cc07 - We rely on several libraries for recreating the test figures at test time. Upgrades to `vdiffr` and `svglite` were causing some problems. Now, we make sure that the same versions of these packages are used to both render the expected figures in tests/figs, as well as render the test-time figures. 

a907a21 - There was still an issue leading to small discrepancies between the expected figures and the actual figs rendered by vdiffr. I could not recreate this issue in several environments, so I had to rely on debug sessions from travis to try to figure out what was causing the small diffs. At the end of the day, it's not an ideal situation b/c you only get 30 mins inside the VM that runs the tests, and 15 mins needs to go to installing ggExtra's deps. So basically you have to keep restarting the build so that you can play around for 15 mins. Moving the build into a docker container fixes the figure discrepancies and makes it easy to debug testing code...I still have no idea were the small discrepancies were coming from, though.

**Comment by crew102** [comment]
If you want I can rebase this branch so it's not so messy

**Comment by daattali** [comment]
Looks like that first commit you mention is my fault! :smiling_imp: 

Moving the build to docker: it sounds like you spent a ton of time trying to debug this, thanks for all that. If this works , I hope you at least got the satisfaction at the end!

**Comment by daattali** [comment]
That worked! Awesome job. Can you explain what the last line in the travis config file does?

**Comment by crew102** [comment]
The last line of the config file just cats the contents of one of the log files (which the `R CMD check` output is written to, at least in part) to stdout. It seemed to me like the majority of the relevant output is already sent to stdout (and thus viewable in the travis job log), but I noticed that R-travis prints out some log files after `R CMD check` is run (e.g., lines 3302, 3789, etc. here: https://travis-ci.org/daattali/ggExtra/builds/297057075?utm_source=github_status&utm_medium=notification), so I thought there might be a reason to include one of the high-level log files.

**Pull request merged**